common/core/desktop/doc/BUILDING.md

```diff
@@ -5,96 +5,129 @@
 - Python 3
 - Meson build system 0.50 or later.
 - C++14 or later compiler (VC++ 2019 or later for Windows).
+- Rust from <https://www.rust-lang.org/tools/install> (or `cargo` package on Linux)
 - lib std::fs
 - kmcomp (for tests) -- must be added to path
 
-## Installing Python3
+### Windows
+
+You can get the official Python installer from the official Python site:
+<https://www.python.org/downloads/windows/>
+
+Ensure you have Python3 correctly installed and can run the command `pip`.
+
+Then install meson:
+
+```bash
+python3 -m pip install meson
+```
+
+Note on paths for kmcomp:
+
+The search path can be edited through System settings / Advanced system settings
+/ Environment Variables / User environment variables.
+
+If you have Keyman Developer installed, add `%KeymanDeveloperPath%` to your
+path. Otherwise, add the path where you extracted the kmcomp archive.
 
 ### Linux
 
-You will be able to install a python3 package in any reputable recent version
-of linux using its package manager if it's not already installed.
+#### Ubuntu and Debian
 
-### Mac OS X
+You can install `meson` and Rust through the package manager:
 
-You can get the official installer from the official Python site:
-<https://www.python.org/downloads/mac-osx/>
+```bash
+sudo apt update
+sudo apt install meson cargo
+```
 
-### Windows
+You'll also need the `kmcomp` wrapper - see below.
 
-You can get the official installer from the official Python site:
-<https://www.python.org/downloads/windows/>
+#### Other Linux distributions
 
-## Installing Meson
+You will be able to install a python3 package in any reputable recent version of
+linux using its package manager if it's not already installed.
 
-Ensure you have Python3 correctly installed and can run the command `pip3`.
+Then install meson:
 
 ```bash
+sudo apt install python3 python3-pip
 python3 -m pip install meson
 ```
 
```

Comment:
```suggestion

## Installing Rust
### Windows 
After downloading and running the executable from:
<https://www.rust-lang.org/tools/install>
Run the command:
```bash
rustup target add i686-pc-windows-msvc
```
this will ensure this target is available. 
```